BothamNetworking/BothamAPIClient.swift

```diff
@@ -115,6 +115,15 @@ public class BothamAPIClient {
         return interceptedResponse
     }
 
+    private func mapNSErrorToBothamAPIClientError(error: NSError) -> BothamAPIClientError {
```

Comment:
We should to move this inside the implementation of NSHTTPClient, here we are exposing the implementation of NSHTTPClient and the error code that you should send in order to notify this kind of error.